### PR TITLE
(PUBLISH 6/28/24 Lunchtime): Docs for watchOS and 7.5.0 release 6/28/24

### PR DESCRIPTION
**Pull request by mbruin-NR**
The documentation changes needed for the watchOS and 7.5.0 release

### CODE REVIEW

File: src/content/docs/mobile-monitoring/new-relic-mobile-ios/watchOS/watchOS-agent-setup.mdx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  - Mobile monitoring
+  - New Relic Mobile iOS
+  - Installation
+  - SwiftUI
+  - WatchOS
+metaDescription: How to properly setup your watchOS project with the New Relic agent.
+freshnessValidatedDate: never
+---
+
+WatchOS projects do not normally contain an ApplicationDelegate or ExtensionDelegate class. These delegates are necessary to configure and start the New Relic agent in a watchOS project. To add an AppDelegate to your watchOS application, follow the steps detailed here. Before adding the AppDelegate, follow the [outlined procedure](/docs/mobile-monitoring/new-relic-mobile-ios/installation/spm-installation) to correctly install the New Relic agent.
```
Comment (akristen) [inline src/content/docs/mobile-monitoring/new-relic-mobile-ios/watchOS/watchOS-agent-setup.mdx:13]: What do these classes do? Why are they a dependency? @mbruin-NR

Comment (mbruin-NR) [reply src/content/docs/mobile-monitoring/new-relic-mobile-ios/watchOS/watchOS-agent-setup.mdx]: They are classes defined by protocols to be used with Apple APIs.
https://developer.apple.com/documentation/watchkit/wkapplicationdelegate
https://developer.apple.com/documentation/watchkit/wkextensiondelegate

Comment (akristen) [reply src/content/docs/mobile-monitoring/new-relic-mobile-ios/watchOS/watchOS-agent-setup.mdx]: So is the purpose of this doc to show customers how to install an agent to a watchOS application, but to be able to do that they need to first add Apple APIs to their apps so that it's compatible with NR?

Comment (akristen) [reply src/content/docs/mobile-monitoring/new-relic-mobile-ios/watchOS/watchOS-agent-setup.mdx]: I'm a little confused by the intro. It seems like they need to first install the agent, then add these classes... but the installation doesn't work _unless_ the APIs are added to their app?

File: src/content/docs/mobile-monitoring/new-relic-mobile-ios/watchOS/watchOS-agent-setup.mdx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+freshnessValidatedDate: never
+---
+
+WatchOS projects do not normally contain an `ApplicationDelegate` or `ExtensionDelegate` class. These delegates are necessary to configure and start the New Relic agent in a watchOS project. Before adding the AppDelegate, follow the [outlined procedure](/docs/mobile-monitoring/new-relic-mobile-ios/installation/spm-installation) to correctly install the New Relic agent. 
+
+## WatchOS Application [#watchOS_App]
+
+To add an `AppDelegate` to your watchOS application, follow the steps:
+
+1. Add a new file to your project named `WatchAppDelegate.swift`
```
Comment (akristen) [inline src/content/docs/mobile-monitoring/new-relic-mobile-ios/watchOS/watchOS-agent-setup.mdx:19]: Does it matter where in their project this file is added? Where should we recommend they put it?

File: src/content/docs/mobile-monitoring/new-relic-mobile-ios/watchOS/watchOS-agent-setup.mdx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+1. Then in that file add a class named `WatchAppDelegate` that inherits from `NSObject` and conforms to the `WKApplicationDelegate` protocol.
```
Comment (akristen) [inline src/content/docs/mobile-monitoring/new-relic-mobile-ios/watchOS/watchOS-agent-setup.mdx:20]: Will our audience understand what to do to "conform to X protocol"? should we elaborate on that?

File: src/content/docs/mobile-monitoring/new-relic-mobile-ios/watchOS/watchOS-agent-setup.mdx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+1. Add a new file to your project named `WatchAppDelegate.swift`
+1. Then in that file add a class named `WatchAppDelegate` that inherits from `NSObject` and conforms to the `WKApplicationDelegate` protocol. 
+1. In the `WatchAppDelegate` class you created, add the `applicationDidFinishLaunching` function. 
+1. As close to the start of `applicationDidFinishLaunching` as possible add `NewRelic.start(withApplicationToken: "APP_TOKEN")` replacing `APP_TOKEN` with your [application token](/docs/mobile-apps/viewing-your-application-token).
+
+<Callout variant="important">
+     To ensure proper instrumentation, you must start the agent on the first line of `applicationDidFinishLaunching()`, and run the agent on the main thread. Starting the call later, on a background thread, or asynchronously can cause unexpected or unstable behavior.
+ </Callout>
+    
+```swift
```
Comment (akristen) [inline src/content/docs/mobile-monitoring/new-relic-mobile-ios/watchOS/watchOS-agent-setup.mdx:28]: ! I see, is this an example of what a user _might_ get when they follow all of the previous steps?

File: src/content/docs/mobile-monitoring/new-relic-mobile-ios/watchOS/watchOS-agent-setup.mdx [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+
+1. In the main app structure object add the following line: `@WKApplicationDelegateAdaptor var appDelegate: WatchAppDelegate`. 
+
+ ```swift
+import NewRelic
+
+@main
+struct Watch_App: App {
+    @WKApplicationDelegateAdaptor var appDelegate: WatchAppDelegate
```
Comment (akristen) [inline src/content/docs/mobile-monitoring/new-relic-mobile-ios/watchOS/watchOS-agent-setup.mdx:49]: Does this need a closing `}`?

File: src/content/docs/mobile-monitoring/new-relic-mobile-ios/watchOS/watchOS-unsupported-features.mdx [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+tags:
+  - Mobile monitoring
+  - New Relic Mobile iOS
+  - Troubleshoot
+  - watchOS
+metaDescription: 'The features currently unsupported by the watchOS agent'
+freshnessValidatedDate: never
+---
+
+Apple does not provide the all the same APIs to watchOS platform. Because of this some of the features that the iOS agent provides are not available on watchOS. We seek to have parity in our platforms so as Apple releases new functionality we will continue to investigate solutions.
```
Comment (akristen) [inline src/content/docs/mobile-monitoring/new-relic-mobile-ios/watchOS/watchOS-unsupported-features.mdx:13]: This doc could be another section inside the get started with new relic watchOS doc above. What do you think about adding that as another h2?

File: src/content/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/set-custom-user-id.mdx [hunk 7]
```diff
@@ -172,6 +172,14 @@ A user identifier is useful for several reasons. Using the [Crash analysis UI](/
 
 If you do not want to explicitly identify users because of privacy rules, this method is still useful when tracking user segments, such as paid vs. free end users, or registered vs. unregistered users. This allows you to filter or facet on the number and types of crashes experienced by each segment, and do outreach based on that segment.
 
+<Callout variant="important">
+  The following enhancements are added starting iOS agent v7.5.0
+
+  1. When `setUserId(value: string|null)` is called, if `userId` was previously null and new value is not-null, the current session continues, and new `userId` will be set.
+
+  2. When `setUserId(value: string|null)` is called, if `userId`was previously not-null and new `userId` is different including `null`, the current session will be ended, and a harvest will be performed. A new session will be created with the new `userId`.
```
Comment (akristen) [inline src/content/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/set-custom-user-id.mdx:180]:
```suggestion
  1. When `setUserId(value: string|null)` is called, if `userId`was previously not-null and new `userId` is different including `null`, the current session will be ended, and a harvest will be performed. A new session will be created with the new `userId`.
```

**Comment by akristen** [comment]
@mbruin-NR Hi! Thanks so much for getting all this information together. I made some structure changes to this PR directly so you could see in a deploy preview how it would look, but am happy to reverse anything. 

- the biggest change is that I merged the unsupported features doc into the install doc as an h2. I did this because IMO when I first saw an unsupported features doc, I assumed that it was a doc outlining the limitations of the New Relic agent, which was confusing. 
- Given how small that doc was, and since those limitations are due to a misalignment between APIs in iOS vs watchOS, I think this structure change showcases watchOS front and center, and showcases your team + all the hard work y'all did to implement this new feature! 
- In other words, having an "unsupported features doc" almost diminishes the "watchOS agent" until someone reads it, and it's better to show off the feature first and then at the end explain any lack of parity at the bottom.
- From the customer POV, it also gives them all the information they need in one area so they don't lose track of anything. :) (But really, we want to show off your hard work ;))

Otherwise I just made some light copy changes to the install doc. LMK what you think!

File: src/content/docs/mobile-monitoring/new-relic-mobile-ios/watchOS/instrument-watchos.mdx [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+        func applicationDidFinishLaunching() {
+                NewRelic.start(withApplicationToken: "APP_TOKEN")
+                return true
+            }
+        }
+    ```
+
+## Limitations to the watchOS platform [#ios-limitations]
+
+Apple doesn't provide the same iOS APIs to the watchOS platform. This means that certain iOS agent features are not yet available on watchOS. We seek to have parity in our platforms, so as Apple releases new functionality we will continue to investigate solutions. Here are some known, currently unsupported features:
```
Comment (akristen) [inline src/content/docs/mobile-monitoring/new-relic-mobile-ios/watchOS/instrument-watchos.mdx:68]:
```suggestion
Apple doesn't provide the same iOS APIs to the watchOS platform. This means that certain iOS agent features are not yet available on watchOS. We seek to have parity in our platforms, so as Apple releases new functionality we will continue to update the agent. Here are some known, currently unsupported features:
```

File: src/content/docs/mobile-monitoring/new-relic-mobile-ios/watchOS/instrument-watchos.mdx [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+
+* Crash handling: PLCrashReporter is a third party library that the agent uses, which is not compatible with watchOS.
+* Crash count: Because crashes are not reported on watchOS, we can't collect crash count.
+* Handled exceptions: The agent captures and uploads handled exceptions, but New Relic can't currently display them.
+* Background harvesting: If `NRFeatureFlag_BackgroundReporting` is enabled, the agent will continue to monitor the application, but won't harvest or upload to New Relic while in background.
+* Offline marked events: If `NRFeatureFlag_OfflineStorage` is enabled, the agent can store offline harvests, then send them when online. Events created in an offline state, however, won't be marked as such on the attribute level.
+* Hot/Cold app start times: The agent depends on UIKit notifications to capture hot/cold start times, but this information is not yet available for watchOS.
+
+If you need additional help, get support at [support.newrelic.com](https://support.newrelic.com).
```
Comment (akristen) [inline src/content/docs/mobile-monitoring/new-relic-mobile-ios/watchOS/instrument-watchos.mdx:77]:
```suggestion
If you need additional help, you can reach support at [support.newrelic.com](https://support.newrelic.com).
```

File: src/nav/mobile-monitoring.yml [hunk 10]
```diff
@@ -75,11 +75,12 @@ pages:
                 path: /docs/mobile-monitoring/new-relic-mobile-ios/configuration/app-launch-times-ios-apps
               - title: iOS device obfuscation
                 path: /docs/mobile-monitoring/new-relic-mobile-ios/configuration/ios-device-id-obfuscation
-
           - title: tvOS
             pages:
               - title: tvOS compatibility and requirements
                 path: /docs/mobile-monitoring/new-relic-mobile-ios/tvos/new-relic-tvos-compatibility-requirements
+          - title: Instrument your watchOS project with New Relic
```
Comment (akristen) [inline src/nav/mobile-monitoring.yml:82]:
```suggestion
          - title: watchOS
```

Comment (akristen) [reply src/nav/mobile-monitoring.yml:82]: I can make this a watchos folder with watchOS doc in it, similar to tvOS. lmk what you prefer

File: src/content/docs/mobile-monitoring/new-relic-mobile-ios/watchOS/watchOS-agent-setup.mdx (hunk 3 above)
Comment (mbruin-NR) [reply src/content/docs/mobile-monitoring/new-relic-mobile-ios/watchOS/watchOS-agent-setup.mdx]: Protocols are a common part of Apple development

File: src/content/docs/mobile-monitoring/new-relic-mobile-ios/watchOS/watchOS-unsupported-features.mdx (hunk 6 above)
Comment (mbruin-NR) [reply src/content/docs/mobile-monitoring/new-relic-mobile-ios/watchOS/watchOS-unsupported-features.mdx]: That works for me.

File: src/content/docs/mobile-monitoring/mobile-monitoring-ui/logs.mdx  [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+
+In mobile apps, logs can be sent to New Relic.
+
+## APIs to Capture Logs [#logs-apis]
+
+<Callout variant="important">
+### Understanding Mobile Logging delays [#mobile-logging-delays]
+
+It will take up to 15 minutes for logs to appear within New Relics Logs Tab. 
+It will take up to 15 minutes for changes to Mobile Logs toggle, sampling rate, and log level to be reflected within your mobile application.
```
Comment (ywang-nr) [inline src/content/docs/mobile-monitoring/mobile-monitoring-ui/logs.mdx :19]: Configuration update should be within 5 mins. Is there a limitation on iOS side? Please let us know so we can consolidate.

File: src/content/docs/mobile-monitoring/mobile-monitoring-ui/logs.mdx  [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+
+NewRelic.logError(String message)
+
+NewRelic.log(LogLevel logLevel, String message)
+
+NewRelic.logThrowable(LogLevel logLevel, String message, Throwable throwable)
+
+NewRelic.logAttributes(Map<String, Object> attributes)
+
+NewRelic.logAll(Throwable throwable, Map<String, Objet> attributes)
```
Comment (ywang-nr) [inline src/content/docs/mobile-monitoring/mobile-monitoring-ui/logs.mdx :79]: Typo here: Object

File: src/content/docs/mobile-monitoring/mobile-monitoring-ui/logs.mdx  [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+### Understanding Mobile Logging delays [#mobile-logging-delays]
+
+It will take up to 15 minutes for logs to appear within New Relics Logs Tab. 
+It will take up to 15 minutes for changes to Mobile Logs toggle, sampling rate, and log level to be reflected within your mobile application.
+</Callout>
+
+
+Our mobile agents report logs to New Relic. This can be used to monitor the logs being emitted by your application.
+
+iOS agent: To use the logging APIs add the following APIs:
```
Comment (ywang-nr) [inline src/content/docs/mobile-monitoring/mobile-monitoring-ui/logs.mdx :25]: Can we refine two colons into one

File: src/content/docs/mobile-monitoring/mobile-monitoring-ui/logs.mdx  [hunk 14]
```diff
@@ -0,0 +1,109 @@
...
+NewRelic.logAttributes([
+    "logLevel": "WARN",
+    "message": "This is a test message for the New Relic logging system.",
+    "additionalAttribute1": "attribute1",
+    "additionalAttribute2": "attribute2"
+])
+```
+
+
+Android agent: To use the logging APIs add the following APIs:
```
Comment (ywang-nr) [inline src/content/docs/mobile-monitoring/mobile-monitoring-ui/logs.mdx :60]: Can we refine two colons into one

File: src/content/docs/mobile-monitoring/mobile-monitoring-ui/logs.mdx  (hunk 11 above)
Comment (mbruin-NR) [reply src/content/docs/mobile-monitoring/mobile-monitoring-ui/logs.mdx :19]: 15 minutes is our pessimistic max time.

Comment (ywang-nr) [reply src/content/docs/mobile-monitoring/mobile-monitoring-ui/logs.mdx :19]: We will keep it as 15 mins

File: src/content/docs/mobile-monitoring/mobile-monitoring-ui/mobile-logs.mdx [hunk 15]
```diff
@@ -0,0 +1,107 @@
...
+
+NewRelic.logError(String message)
+
+NewRelic.log(LogLevel logLevel, String message)
+
+NewRelic.logThrowable(LogLevel logLevel, String message, Throwable throwable)
+
+NewRelic.logAttributes(Map<String, Object> attributes)
+
+NewRelic.logAll(Throwable throwable, Map<String, Objet> attributes)
```
Comment (cdillard-NewRelic) [inline src/content/docs/mobile-monitoring/mobile-monitoring-ui/mobile-logs.mdx:79]: Typo here: Object

File: src/content/docs/mobile-monitoring/mobile-monitoring-ui/mobile-logs.mdx [hunk 16]
```diff
@@ -0,0 +1,107 @@
...
+1. In New Relic, navigate to your mobile app.
+2. In the left pane under <DoNotTranslate>**Settings**</DoNotTranslate>, click <DoNotTranslate>**Application**</DoNotTranslate>.
+3. To change the sample rate select a new value in the field under **Sample rate of total sessions**.
+4. To change the log level select your preferred log level in the drop down under **Logs verbosity**. The debug log level should only be used if you want to see agent debugging logs.
+
+## View logs in New Relic [#view-logs-in-new-relic]
+
+To view your logs in the UI:
+
+1. Go to <DoNotTranslate>**[one.newrelic.com > All capabilities](https://one.newrelic.com/all-capabilities)</DoNotTranslate>.
```
Comment (cdillard-NewRelic) [inline src/content/docs/mobile-monitoring/mobile-monitoring-ui/mobile-logs.mdx:104]: Missing closing ** to bold this.

**Review by cdillard-NewRelic — APPROVED** [review]
👍

**Review by rhetoric101 — DISMISSED** [review]
@mbruin-NR and @cdillard-NewRelic thanks for adding these new pages!

**Review by cdillard-NewRelic — APPROVED** [review]

**Review by rhetoric101 — APPROVED** [review]
@cdillard-NewRelic and @mbruin-NR thanks for your work on these docs! These latest versions look good!
